modules/ingest-grok/src/test/java/org/elasticsearch/ingest/grok/GrokProcessorFactoryTests.java

```diff
@@ -59,42 +59,54 @@ public void testBuildMissingField() throws Exception {
         }
     }
 
-    public void testBuildMissingPattern() throws Exception {
+    public void testBuildMissingPatterns() throws Exception {
         GrokProcessor.Factory factory = new GrokProcessor.Factory(Collections.emptyMap());
         Map<String, Object> config = new HashMap<>();
         config.put("field", "foo");
         try {
             factory.create(config);
             fail("should fail");
         } catch (ElasticsearchParseException e) {
-            assertThat(e.getMessage(), equalTo("[pattern] required property is missing"));
+            assertThat(e.getMessage(), equalTo("[patterns] required property is missing"));
         }
+    }
 
+    public void testBuildEmptyPatternsList() throws Exception {
+        GrokProcessor.Factory factory = new GrokProcessor.Factory(Collections.emptyMap());
+        Map<String, Object> config = new HashMap<>();
+        config.put("field", "foo");
+        config.put("patterns", Collections.emptyList());
+        try {
+            factory.create(config);
+            fail("should fail");
+        } catch (ElasticsearchParseException e) {
+            assertThat(e.getMessage(), equalTo("[patterns] List of patterns must not be empty"));
+        }
```

Comment:
oh that's a neat little helper. Is this meant to replace all uses of the old pattern? If so, I would think it would make sense to update all of the test cases, not just this one. what do you think?